specs/dwritecore/DWriteCore.md

```diff
@@ -1235,3 +2025,906 @@ DWRITE_BEGIN_INTERFACE(IDWriteFactory8, "202D44F0-84AC-4D6F-B8D0-704B27675CAA")
+/// with a color glyph. This is passed as an output parameter to various IDWritePaintReader
+/// methods.
+/// </summary>
+/// <remarks>
+/// For a detailed description of how paint elements should be rendered, see the OpenType COLR
+/// table specification. Comments below reference the COLR paint record formats associated with
+/// each paint type.
+/// </remarks>
+struct DWRITE_PAINT_ELEMENT
+{
+    /// <summary>
+    /// Specifies the paint type, and thus which member of the union is valid.
+    /// </summary>
+    DWRITE_PAINT_TYPE paintType;
+
+    /// <summary>
+    /// Specifies type-specific properties of the paint element.
+    /// </summary>
+    union PAINT_UNION
+    {
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_LAYERS.
+        /// Contains one or more child paint elements to be drawn in bottom-up order.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a PaintColrLayers record in the OpenType COLR table.
+        /// Or it may correspond to a BaseGlyph record defined by COLR version 0.
+        /// </remarks>
+        struct PAINT_LAYERS
+        {
+            /// <summary>
+            /// Number of child paint elements in bottom-up order. Use the IDWritePaintReader
+            /// interface's MoveFirstChild and MoveNextSibling methods to retrieve the child paint
+            /// elements. Use the MoveParent method to return to the parent element.
+            /// </summary>
+            UINT32 childCount;
+        } layers;
+
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_SOLID_GLYPH.
+        /// Specifies a glyph with a solid color fill.
+        /// This paint element has no child elements.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a combination of two paint records in the OpenType COLR table:
+        /// a PaintGlyph record, which references either a PaintSolid or PaintVarSolid record.
+        /// Or it may correspond to a Layer record defined by COLR version 0.
+        /// </remarks>
+        struct PAINT_SOLID_GLYPH
+        {
+            /// <summary>
+            /// Glyph index defining the shape to be filled.
+            /// </summary>
+            UINT32 glyphIndex;
+
+            /// <summary>
+            /// Glyph color used to fill the glyph shape.
+            /// </summary>
+            DWRITE_PAINT_COLOR color;
+        } solidGlyph;
+
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_SOLID.
+        /// Specifies a solid color used to fill the current shape or clip.
+        /// This paint element has no child elements.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a PaintSolid or PaintVarSolid record in the OpenType COLR table.
+        /// </remarks>
+        typedef DWRITE_PAINT_COLOR PAINT_SOLID;
```

Comment:
Does this typedef really buy us anything? It's only used below. I suggest you just define solid as a DWRITE_PAINT_COLOR.

Reply:
You're right. This doesn't really buy anything. Same with transform.